apcd-cms/src/apps/submissions/views.py

```diff
@@ -50,8 +50,30 @@ def get_context_data(self, *args, **kwargs):
 
 
         context['header'] = ['Received', 'File Name', ' ', 'Outcome', 'Status', 'Last Updated', 'Actions']
-
-        context.update(paginator(self.request, submission_content))
+        context['filter_options'] = ['All', 'In Process', 'Complete']
+        context['sort_options'] = {'newDate': 'Date: Newest to Oldest', 'oldDate': 'Date: Oldest to Newest'}
+
+        filter = self.request.GET.get('filter')
+        dateSort = self.request.GET.get('sort')
+
+        def getDate(row):
+            date = row['received_timestamp']
+            return date if date is not None else parser.parse('1-1')
+
+        if dateSort is not None:
+            context['selected_sort'] = dateSort
+            submission_content[offset:offset + limit] = sorted(submission_content[offset:offset + limit], key=lambda row:getDate(row), reverse=(dateSort == 'newDate'))
+
+        context['selected_filter'] = None
+        if filter is not None and filter != 'All':
+            context['selected_filter'] = filter
+            submission_content[offset:offset + limit] = table_filter(filter, submission_content[offset:offset + limit], 'status')
```

Comment:
The `table_filter` function doesn't seem to play well with `None`, in the example data provided. It seems like there's some real possibilities of running into a `NoneType` in these records, so the `table_filter` could use a check. I'd recommend breaking it out because that function is getting complicated as a one-liner.